### PR TITLE
[tune] demo exporting trained models in pbt examples

### PR DESCRIPTION
**Pull request by hhbyyh**
For pbt users, it's important to access the trained models after tuning.  Add the demo for exporting models in pbt examples since:
1. Loading from checkpoints in the current code is not robust.
2. Need some code example to demo the usage of exporting model API.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: python/ray/tune/examples/pbt_convnet_example.py [hunk 1]
```diff
@@ -51,6 +52,14 @@ def _save(self, checkpoint_dir):
     def _restore(self, checkpoint_path):
         self.model.load_state_dict(torch.load(checkpoint_path))
 
+    def _export_model(self, export_formats, export_dir):
+        if export_formats == [ExportFormat.MODEL]:
+            path = export_dir + "/exported_convnet.pt"
```
Comment (richardliaw) [inline python/ray/tune/examples/pbt_convnet_example.py]:
```suggestion
            path = os.path.join(export_dir, "/exported_convnet.pt")
```

**Review by richardliaw — APPROVED** [review]
Changes look pretty good. I'm wondering if this is the best way to export a trained model.

**Comment by richardliaw** [comment]
BTW, what does `Loading from checkpoints in the current code is not robust.` mean?

File: python/ray/tune/examples/pbt_convnet_example.py (hunk 1 above)
Comment (hhbyyh) [reply python/ray/tune/examples/pbt_convnet_example.py]: Thanks. Updated.

**Comment by hhbyyh** [comment]
Oh I thought _export_model is designed for that purpose. 
Two concerns about the checkpoint:
1. If we didn't set checkpoint at end, will there always be a checkpoint after tuning?
2. The path to the checkpoint folder is not stable, it's affected by training iterations and future changes. Comparably, a path specified by user seems better.

**Comment by hhbyyh** [comment]
Updated with stopper. I know we can keep the best k models in the checkpoints. But I'm not sure what's the best programming way to load the best models. If we use 
```
os.path.join(analysis.get_best_logdir(), "checkpoint_someiter")
```
Yet it seems too complicated as we need to find the checkpoint dir name.

**Comment by richardliaw** [comment]
Maybe we can introduce a `analysis.get_checkpoints(trial)`?

**Comment by richardliaw** [comment]
jenkins test tune

**Comment by hhbyyh** [comment]
I definitely support adding `analysis.get_checkpoints(trial)`, as it clearly directs the users how to fetch the saved checkpoints and allow us to hide details for future changes.

Right now I'm using 
```
from ray.tune.checkpoint_manager import Checkpoint

best_trial = analysis.get_best_trial("mean_accuracy")
cp = best_trial.checkpoint_manager.best_checkpoints()
paths = [(c.value, c.result['mean_accuracy']) for c in cp if c.storage==Checkpoint.DISK]
```
Let me know if there's another recommended way. 

In some cases the checkpoints can be a combination of MEMORY and DISK checkpoints, we may add an optional parameter to the function. `def get_checkpoints(trial, disk_only=True)`?

**Pull request merged**